src/PackageVersions/FallbackVersions.php

```diff
@@ -54,8 +55,14 @@ public static function getVersion(string $packageName) : string
      */
     private static function getComposerLockPath() : string
     {
-        // bold assumption, but there's not here to fix everyone's problems.
-        $checkedPaths = [__DIR__ . '/../../../../../composer.lock', __DIR__ . '/../../composer.lock'];
+        $checkedPaths = [
+            // The top-level project's ./vendor/composer/installed.json
+            getcwd() . '/vendor/composer/installed.json',
```

Comment:
Ah, careful! `installed.json` and `composer.lock` have different contained data, so this won't work. `installed.json` is an `array`, whereas `composer.lock` is a `map`

Reply:
Good catch.